app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -5,7 +5,9 @@ class ManageIQ::Providers::Redhat::InfraManager::Vm < ManageIQ::Providers::Infra
   include_concern 'ManageIQ::Providers::Redhat::InfraManager::VmOrTemplateShared'
 
   supports :migrate do
-    unless ext_management_system.supports_migrate?
+    if !active?
+      unsupported_reason_add(:migrate, 'VM is already retired')
```

Comment:
Think you should keep the exact same conditions and message as a normal vm_or_template and just add the RHV check, so
```
if blank? || orphaned? || archived?
        unsupported_reason_add(:migrate, "Migrate operation in not supported.")
elsif !ext_management_system.supports_migrate?
```